@here/harp-mapview/lib/text/TextStyleCache.ts

```diff
@@ -54,12 +54,14 @@ export const DEFAULT_TEXT_STYLE_CACHE_ID = "Default";
  *
  * @returns [[TextStyle]] id.
  */
-export function computeStyleCacheId(
-    datasourceName: string,
+function computeStyleCacheId(
+    tile: Tile,
     technique: Technique & Partial<IndexedTechniqueParams>,
     zoomLevel: number
 ): string {
-    return `${datasourceName}_${technique._key}_${zoomLevel}`;
+    const datasourceName = tile.dataSource.name;
+    const key = `${datasourceName}_${tile.tileKey.toHereTile()}_${technique._index}_${zoomLevel}`;
```

Comment:
The cache in proposed this form doesn't make sense at all.

Currently, this is global cache that is reused across tiles in scope of one datasource. Apart from big cost of creating ana maintaining keys, we at least reuse styles across tiles.

With this patch, it's still global cache with global key, but there is mn reuse outside of one particular instance of Tile, so it's just hundreds of unnecessary string concatenations where, this cache could be just direct Tile member:

```
textRenderStyleCache: Map<technique index as number, Text*Style>.
```

(zoom level can be removed if cache is bound to Tile)

Reply:
Good point, I agree. I will update the change.

Reply:
Followup in #1338